web3/types.py

```diff
@@ -295,7 +295,7 @@ class GethSyncingSubscriptionResponse(SubscriptionResponse):
 
 
 class RPCResponse(TypedDict, total=False):
-    error: Union[RPCError, str]
```

Comment:
Note: This change is also breaking but this is how it should be defined. The only reason we allowed `str` is bc of how `EthereumTesterProvider` returned errors but that should not be allowed imo.